test/DynamoCoreWpfTests/CoreUITests.cs

```diff
@@ -708,5 +712,72 @@ public void TestDraggedNode()
             Assert.AreEqual(-8, locatable.X);
             Assert.AreEqual(40, locatable.Y);
         }
+
+        [Test]
+        [Category("UnitTests")]
+        public void WorkspaceContextMenu_TestIfOpenOnRightClick()
+        {
+            var currentWs = View.ChildOfType<WorkspaceView>();
+            Assert.IsNotNull(currentWs, "DynamoView does not have any WorkspaceView");
+            RightClick(currentWs.zoomBorder);
+
+            Assert.IsTrue(currentWs.ContextMenuPopup.IsOpen);
+            // for not throwing 'System.Runtime.InteropServices.InvalidComObjectException' in PresentationCore.dll
+            System.Windows.Threading.Dispatcher.CurrentDispatcher.InvokeShutdown();
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void WorkspaceContextMenu_TestIfNotOpenOnNodeRightClick()
+        {
+            var currentWs = View.ChildOfType<WorkspaceView>();
+            Assert.IsNotNull(currentWs, "DynamoView does not have any WorkspaceView");
+            CreateNodeOnCurrentWorkspace();
+
+            DispatcherUtil.DoEvents();
+            var node = currentWs.ChildOfType<NodeView>();
+            RightClick(node);
+
+            // workspace context menu shouldn't be open
+            Assert.IsFalse(currentWs.ContextMenuPopup.IsOpen);
+
+            // for not throwing 'System.Runtime.InteropServices.InvalidComObjectException' in PresentationCore.dll
+            System.Windows.Threading.Dispatcher.CurrentDispatcher.InvokeShutdown();
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void WorkspaceContextMenu_TestIfInCanvasSearchHidesOnOpeningContextMenu()
+        {
+            var currentWs = View.ChildOfType<WorkspaceView>();
+
+            // show in-canvas search
+            ViewModel.CurrentSpaceViewModel.ShowInCanvasSearchCommand.Execute(ShowHideFlags.Show);
+            Assert.IsTrue(currentWs.InCanvasSearchBar.IsOpen);
+
+            // open context menu
+            RightClick(currentWs.zoomBorder);
+
+            Assert.IsTrue(currentWs.ContextMenuPopup.IsOpen);
+            Assert.IsFalse(currentWs.InCanvasSearchBar.IsOpen);
```

Comment:
Yes, Masha I understand what you put there, but my point is that at this moment, the inCanvasSearchBar(within the popup menue is also open), so Assert.IsFalse(currentWs.InCanvasSearchBar.IsOpen); will fail. I have merged your change and NUnit actually comfirmed that this unit test will fail. So my point is that we do not need this unit test in this case because in both case, the searchbar.IsOpen will be true. Please confirm and do another submission.